src/redux/mission/missionApi.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-fallthrough */
+import axios from 'axios';
+
+const url = 'https://api.spacexdata.com/v3/missions';
+
+const initialState = [];
+
+const FETCH_MISSIONS = 'FETCH_MISSIONS';
+const JOIN_MISSION = 'JOIN_MISSION';
+const missionReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_MISSIONS:
+      return [...action.fetchedMissions];
+    case JOIN_MISSION:
+      return state.map((mission) => (mission.id === action.payload
+        ? { ...mission, status: !mission.status }
+        : mission));
+
+    default:
+      return state;
+  }
+};
+export const joinMission = (payload) => ({
+  type: 'JOIN_MISSION',
+  payload,
+});
+
+export const getMissions = () => async (dispatch) => {
+  const result = await axios.get(url);
+  const missions = result.data;
+  const fetchedMissions = [];
+  missions.forEach((mission) => {
+    const id = mission.mission_id;
+    const name = mission.mission_name;
+    const desc = mission.description;
+    const status = false;
+    fetchedMissions.push({
+      id, name, desc, status,
+    });
+  });
+  dispatch({
+    type: FETCH_MISSIONS,
+    fetchedMissions,
+  });
+};
+
```

Comment:
- [ ] Even though the implementation is working perfectly, a bunch of code is still in one file. It's not helpful at all. Let us split the code into separate files so that we can make the component smaller.

      -  Let us create three folders under the redux folder
      
- Actions folder will keep the `action creators and action types`   

Ex of action creators: `export const allowLeaveMission = (payload) => ({
  type: LEAVE_MISSION,
  payload,
});   `

Example of action type: `const JOIN_MISSION = 'space/missions/JOIN_MISSION';`

- Reducers folder will keep the cases 

Example:

`const missionsReducer = (state = { missions: [] }, action) => {
  switch (action.type) {
    case MISSION_START:
      return {
        ...state,
        loading: true,
      };
    case MISSION_FAILED:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };`
      
- thunk folder will keep the API implementations, fetch some information from the API link.
 
Example

`export const loadMissionsData = () => (dispatch) => {
  dispatch(loadingStart());
  fetchData()
    .then((res) => res.json())
    .then((result) => dispatch(loadMission(result)))
    .catch((err) => dispatch(manageFailure(err.message)));
};`

Please, let us try to split it out into different files to keep our code clean and easy to maintain.